akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,606 @@
-
-  def guardian(outstanding: Map[ActorRef[_], ActorRef[Status]] = Map.empty): Behavior[GuardianCommand] =
-    Behaviors.immutable[GuardianCommand] {
-      case (ctx, r: RunTest[t]) ⇒
-        val test = ctx.spawn(r.behavior, r.name)
-        ctx.schedule(r.timeout, r.replyTo, Timedout)
-        ctx.watch(test)
-        guardian(outstanding + ((test, r.replyTo)))
-      case (_, Terminate(reply)) ⇒
-        reply ! Success
-        stopped
-      case (ctx, c: Create[t]) ⇒
-        c.replyTo ! ctx.spawn(c.behavior, c.name)
-        same
-    } onSignal {
-      case (ctx, t @ Terminated(test)) ⇒
-        outstanding get test match {
-          case Some(reply) ⇒
-            t.failure match {
-              case None     ⇒ reply ! Success
-              case Some(ex) ⇒ reply ! Failed(ex)
-            }
-            guardian(outstanding - test)
-          case None ⇒ same
-        }
-      case _ ⇒ same
-    }
 }
 
-abstract class ActorContextSpec extends TypedAkkaSpec {
-  import ActorContextSpec._
+abstract class ActorContextSpec extends ActorTestKit with TypedAkkaSpecWithShutdown {
 
-  implicit val timeout = Timeout(3.seconds)
+  import ActorSpecMessages._
 
-  val config = ConfigFactory.parseString(
-    """
-     akka {
-       loglevel = WARNING
-       loggers = ["akka.testkit.TestEventListener"]
-       actor.debug {
-         lifecycle = off
-         autoreceive = off
-       }
-     }""")
+  def decoration[T]: Behavior[T] ⇒ Behavior[T]
 
-  implicit lazy val system: ActorSystem[GuardianCommand] =
-    ActorSystem(guardian(), AkkaSpec.getCallerName(classOf[ActorContextSpec]), config = Some(config withFallback AkkaSpec.testConf))
-
-  val expectTimeout = 3.seconds
-  import AskPattern._
-
-  implicit def scheduler = system.scheduler
-
-  lazy val blackhole = await(system ? Create(immutable[Any] { case _ ⇒ same }, "blackhole"))
-
-  override def afterAll(): Unit = {
-    Await.result(system.terminate, timeout.duration)
+  implicit class BehaviorDecorator[T](behavior: Behavior[T]) {
+    def decorate: Behavior[T] = decoration(behavior)
   }
 
-  // TODO remove after basing on ScalaTest 3 with async support
-  import akka.testkit._
-
-  def await[T](f: Future[T]): T = Await.result(f, timeout.duration * 1.1)
-
-  /**
-   * Run an Actor-based test. The test procedure is most conveniently
-   * formulated using the [[StepWise]] behavior type.
-   */
-  def runTest[T: ClassTag](name: String)(behavior: Behavior[T])(implicit system: ActorSystem[GuardianCommand]): Future[Status] =
-    system ? (RunTest(name, behavior, _, timeout.duration))
+  "An ActorContext" must {
 
-  // TODO remove after basing on ScalaTest 3 with async support
-  def sync(f: Future[Status])(implicit system: ActorSystem[GuardianCommand]): Unit = {
-    def unwrap(ex: Throwable): Throwable = ex match {
-      case ActorInitializationException(_, _, ex) ⇒ ex
-      case other                                  ⇒ other
-    }
+    "converge in cyclic behavior" in {
```

Comment:
I think the old name for this test was better, as canonicalize is the internal term for taking a behavior that can either be "symbolical" (stopped, same, ignore), deferred or an actual behavior and turn it into an actual runnable behavior.